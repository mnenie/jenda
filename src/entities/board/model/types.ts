import type { User } from '@/entities/user/model';

type DateParams = {
  createdAt?: Date;
  updatedAt?: Date;
  deletedAt?: Date;
};

export type Status = 'work' | 'archive' | 'closed' | 'not active';

export interface BoardPreview extends DateParams {
  _id: string;
  name: string;
  description?: string;
  users: User[];
}

export interface Board extends BoardPreview, DateParams {
  columns?: Column[];
  color?: string;
  status: Status;
}

type Tag = {
  _id: string;
  name: string;
};

export interface Card extends DateParams {
  _id: string;
  title: string;
  priority: 'none' | 'low' | 'medium' | 'high';
  tags?: Tag[];
  chat?: boolean;
  chatCount?: number;
  users: User[];
}

export interface Column extends DateParams {
  _id: string;
  title: string;
  cards?: Card[];
}

export interface StatusBadge {
  _id: string;
  indicator: string;
  status: Status;
}
