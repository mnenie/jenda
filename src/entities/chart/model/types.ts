import type { Component } from 'vue';

export interface Chart {
  id: string;
  titleKeyI18n: string;
  descriptionKeyI18n: string;
  chart: Component;
}
