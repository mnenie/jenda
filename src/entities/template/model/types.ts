export interface Template {
  id: string;
  title: string;
  tag: string;
  description: string;
  date: string;
  user: string;
}
