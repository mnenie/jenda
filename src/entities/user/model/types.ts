type DateParams = {
  createdAt?: Date;
  updatedAt?: Date;
  deletedAt?: Date;
};

export interface UserType extends DateParams {
  _id?: string;
  email: string;
  photoUrl?: string;
}

export interface UserAuth extends UserType {
  token: string;
}
