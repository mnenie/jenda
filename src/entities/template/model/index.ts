export * from './types';
export * from './templates';
