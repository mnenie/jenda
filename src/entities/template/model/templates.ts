import type { Template } from '../model';

export const templatesInfo: Template[] = [
  {
    id: '0',
    title: 'Base Kanban',
    img: 'https://storage.weeek.net/templates/tm/development-and-product/scrum.png',
    tag: 'Recommended',
    description: `Create a basic project with "Base Kanban" template`,
    date: 'June 12, 2024',
    user: 'https://avatars.githubusercontent.com/u/121057011?v=4'
  },
  {
    id: '1',
    title: 'Roadmap',
    img: 'https://storage.weeek.net/templates/tm/development-and-product/roadmap.png',
    tag: 'New',
    description: `Create a project based on our "Roadmap" template`,
    date: 'June 18, 2024',
    user: 'https://avatars.githubusercontent.com/u/121057011?v=4'
  },
  {
    id: '2',
    title: 'Web Development',
    img: 'https://storage.weeek.net/templates/tm/development-and-product/web-development.png',
    tag: 'New',
    description: `Create a basic project with "Web Development" template`,
    date: 'July 16, 2024',
    user: 'https://avatars.githubusercontent.com/u/121057011?v=4'
  }
];
