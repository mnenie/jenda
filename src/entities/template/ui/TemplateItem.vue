<script setup lang="ts">
import { UiBadge } from '@/shared/ui';
import type { Template } from '../model';

defineProps<{
  template: Template;
}>();
</script>

<template>
  <div :class="$style.item">
    <div :class="$style.img_wrapper">
      <img :src="template.img" :alt="template.title" />
    </div>
    <div :class="$style.main_content">
      <UiBadge :variant="'secondary'" style="margin-bottom: 12px">{{ template.tag }}</UiBadge>
      <div :class="$style.text">
        <p class="text-base" style="font-weight: 500">{{ template.title }}</p>
        <span>{{ template.description }}</span>
      </div>
      <div :class="$style.bottom">
        <span class="text-xs">{{ template.date }}</span>
        <img v-tooltip.bottom-end="'creator'" :src="template.user" />
      </div>
    </div>
  </div>
</template>

<style module lang="scss">
@import '@/app/styles/mixins';

.item {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  background-color: white;
  border: 1px solid var(--zinc-200);
  width: 100%;
  height: fit-content;
  border-radius: 10px;
  cursor: pointer;
  align-items: flex-start;
  @include transition;

  &:hover {
    border: 1px solid var(--purple-main);
  }

  .img_wrapper {
    width: 100%;
    height: 130px;
    border-radius: 10px 10px 0 0;
    border-bottom: 1px solid var(--zinc-200);

    & img {
      width: 100%;
      height: 100%;
      object-fit: cover;
    }
  }
  .main_content {
    padding: 15px;

    .text {
      display: flex;
      flex-direction: column;
      gap: 2px;
      margin-bottom: 15px;
    }

    & span {
      color: var(--zinc-500);
    }
  }

  .bottom {
    display: flex;
    align-items: center;
    justify-content: space-between;

    & img {
      width: 20px;
      height: 20px;
      border-radius: 50%;
    }
  }
}

:global(html.dark) {
  .item {
    background-color: rgba(var(--zinc-rgb-600), 0.4);
    border-color: var(--zinc-600);
    &:hover {
      border-color: var(--zinc-400);
    }
    .img_wrapper{
      border-color: var(--zinc-600);
    }
    .main_content{
      & span {
        color: var(--zinc-200);
      }
    }
  }
}
</style>
