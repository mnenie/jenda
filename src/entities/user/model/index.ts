export * from './user.store';
export * from './types';
