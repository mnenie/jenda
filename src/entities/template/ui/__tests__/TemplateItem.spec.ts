import { describe, it, expect } from 'vitest';
import { shallowMount } from '@vue/test-utils';
import '@/shared/lib/vitest-utils/cookiesI18n-mock';
import i18n from '@/shared/lib/i18n';
import TemplateItem from '../TemplateItem.vue';
import { templatesInfo } from '../../model';

describe('tests for TemplateItem.vue ', () => {
  const wrapper = shallowMount(TemplateItem, {
    global: {
      plugins: [i18n],
      mocks: {
        t: (key: string) => {
          const translations: Record<string, string> = {
            'templates.user': 'test user'
          };
          return translations[key];
        }
      }
    },
    props: {
      template: templatesInfo[0]
    }
  });

  it('should render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
