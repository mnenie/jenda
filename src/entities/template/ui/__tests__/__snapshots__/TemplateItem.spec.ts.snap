// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`tests for TemplateItem.vue  > should render correctly 1`] = `
"<div class="item">
  <div class="main_content">
    <ui-badge-stub variant="outline" style="margin-bottom: 12px;"></ui-badge-stub>
    <div class="text">
      <p class="text-sm">Base Kanban</p><span class="text-xs">Create a basic project with "Base Kanban" template</span>
    </div>
    <div class="bottom"><span class="text-xs">June 12, 2024</span><img src="https://avatars.githubusercontent.com/u/121057011?v=4"></div>
  </div>
</div>"
`;
